src/porepy/numerics/time_step_control.py

```diff
@@ -1,130 +1,191 @@
                 "Initial time step cannot be larger than final simulation time."
             )
-        elif dt_init < dt_min_max[0]:
-            raise ValueError(
-                "Initial time step cannot be smaller than minimum time step."
-            )
-        elif dt_init > dt_min_max[1]:
-            raise ValueError(
-                "Initial time step cannot be larger than maximum time step."
-            )
 
-        # NOTE: The above checks satisfy that minimum time step <= maximum time step
-
-        # Check maximum number of iterations. Note that 0 is a possibility. This
-        # will imply that the solver reaches convergence directly, i.e., as in
-        # linear solvers.
-        if iter_max < 0:
-            raise ValueError("Maximum number of iterations must be non-negative.")
-
-        # Checks for optimal iteration range
-        if iter_optimal_range[0] > iter_optimal_range[1]:
-            s = "Lower optimal iteration range cannot be larger than"
-            s += " upper optimal iteration range."
-            raise ValueError(s)
-        elif iter_optimal_range[1] > iter_max:
-            s = "Upper optimal iteration range cannot be larger than"
-            s += " maximum number of iterations."
-            raise ValueError(s)
-        elif iter_optimal_range[0] < 0:
-            s = "Lower optimal iteration range cannot be negative."
-            raise ValueError(s)
-
-        # Checks for lower and upper multiplication factors
-        if iter_lowupp_factor[0] <= 1:
-            raise ValueError("Expected lower multiplication factor > 1.")
-        elif iter_lowupp_factor[1] >= 1:
-            raise ValueError("Expected upper multiplication factor < 1.")
-
-        # Checks for sensible combinations of iter_optimal_range and iter_lowupp_factor
-        if dt_min_max[0] * iter_lowupp_factor[0] > dt_min_max[1]:
-            raise ValueError("Encountered dt_min * iter_low_factor > dt_max.")
-        elif dt_min_max[1] * iter_lowupp_factor[1] < dt_min_max[0]:
-            raise ValueError("Encountered dt_max * iter_upp_factor < dt_min.")
-
-        # Check for recomputation factor
-        if recomp_factor >= 1:
-            raise ValueError("Expected recomputation multiplication factor < 1.")
-
-        # Check for maximum number of recomputation attempts
-        if recomp_max <= 0:
-            raise ValueError("Number of recomputation attempts must be > 0.")
+        # Set dt_min_max if necessary
+        if dt_min_max is None:
+            dt_min_max = (0.001 * schedule[-1], 0.1 * schedule[-1])
+
+        # More sanity checks below. Note that all the remaining sanity checks (but one) are
+        # only needed when constant_dt = False. Thus, to save time when constant_dt = True,
+        # we use this rather ugly if-statement from below.
+
+        if not constant_dt:
+
+            # Sanity checks for dt_min and dt_max
+            if dt_init < dt_min_max[0]:
+                raise ValueError(
```

Comment:
```suggestion
                # NOTE: If dt_min_max was not set on initialization, a value is assigned besed on the total
                # simulation time.  This may unwittingly trigger the below error.
                raise ValueError(
```
I suggest adding a comment like this to give users something to work with if they unwittingly trigger this error. Someone will appreciate it.

Reply:
You are right, I did not consider this possibility. If the user passes any `dt_min < 0.001 schedule[-1]`, and don't pass `dt_min_max`, then this error will be triggered.

I will raise a costume error for this case, suggesting to pass `dt_min_max` to avoid getting an error triggered by `dt_min < dt_min_max[0]`.